pkg/expr/sql/db.go

```diff
@@ -1,22 +1,51 @@
 package sql
 
 import (
-	"errors"
+	"context"
 
+	sqle "github.com/dolthub/go-mysql-server"
+	mysql "github.com/dolthub/go-mysql-server/sql"
+
+	"github.com/dolthub/go-mysql-server/sql/analyzer"
 	"github.com/grafana/grafana-plugin-sdk-go/data"
 )
 
-type DB struct {
-}
+// DB is a database that can execute SQL queries against a set of Frames.
+type DB struct{}
 
-func (db *DB) RunCommands(commands []string) (string, error) {
-	return "", errors.New("not implemented")
-}
+// QueryFrames runs the sql query query against a database created from frames, and returns the frame.
+// The RefID of each frame becomes a table in the database.
+// It is expected that there is only one frame per RefID.
+// The name becomes the name and RefID of the returned frame.
+func (db *DB) QueryFrames(name string, query string, frames []*data.Frame) (*data.Frame, error) {
+	pro := NewFramesDBProvider(frames)
+	session := mysql.NewBaseSession()
+	ctx := mysql.NewContext(context.Background(), mysql.WithSession(session))
```

Comment:
should context be passed in here so that processing can be stopped if the parent request is cancelled?
```suggestion
func (db *DB) QueryFrames(ctx context.Context, name string, query string, frames []*data.Frame) (*data.Frame, error) {
	pro := NewFramesDBProvider(frames)
	session := mysql.NewBaseSession()
	ctx := mysql.NewContext(ctx, mysql.WithSession(session))
```